src/Tasks/CultureInfoCache.cs

```diff
@@ -56,6 +57,21 @@ static HashSet<string> InitializeValidCultureNames()
         /// <returns>True if the culture is determined to be valid.</returns>
         internal static bool IsValidCultureString(string name)
         {
+#if NET5_0_OR_GREATER
+            if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_4))
+            {
+                try
+                {
+                    // GetCultureInfo throws if the culture doesn't exist
+                    CultureInfo.GetCultureInfo(name, true);
```

Comment:
This essentially says "skip the hardcoded list on net6," right? Why is that correct?

Reply:
The hardcoded list has always been a workaround to the fact that we didn't have an API that would specifically tell us when a culture was valid or not (`CultureInfo.GetCultures(AllCultures)` wasn't made for verification purposes). It's not ideal to use the list at all, and this avoids it because starting in net5.0 this `GetCultureInfo(name, true)` was introduced for specifically this purpose.

The only valid case for us to use the list moving foward would be in net472 for backwards compat when some culture or alias isn't supported. But net472 happens to work because on Windows the API we're currently using happens to contain the cultures that aren't included on linux.

Reply:
Does the back compat issue not apply to net6 as well?

Reply:
it does, `NET5_0_OR_GREATER` is defined in the SDK and would apply to net5.0+. It happens around here:

https://github.com/dotnet/sdk/blob/dd33b56b1c8ab92cef9fd4173585b9915003e0b8/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L207